frontend/benefit/applicant/src/constants.ts

```diff
@@ -185,3 +187,9 @@ export enum VALIDATION_MESSAGE_KEYS {
 }
 
 export const dateRegex = /(?:(?:0?[1-9]|1\d|2\d)\.(?:0?[1-9]|1[0-2])|(?:30\.(?!02)(?:0[1-9]|1[0-2]))|(?:31\.(?:0[13578]|1[02])))\.(?:19|20)\d{2}/;
```

Comment:
Where are these regex coming from? Perhaps link to source would help. Also you have to communicate to backend coders that they will use the same validation when sending application to backend.

Reply:
i think it was my regex of dd/mm/yyyy, i definitely was testing couple of implementations and took most reliable one, but i cannot find the link. Regex is always the matter of improvising and improving, and there is no "correct" way, but i agree that backend and frontend should have the same rules.

Reply:
Some comment line to explain it would be good, at least to tell what is the expected date format

Reply:
It seems to be dd.mm.yyyy. Some corner cases like leap day cannot be validated with just regex so it needs some additional validation.

Reply:
you can actually disable direct typing to the hds datepicker (if its supported). then you cannot select the wrong date.